src/main/java/com/arpnetworking/metrics/mad/model/json/Version2g.java

```diff
@@ -0,0 +1,334 @@
+
+/**
+ * Model for the version 2g query log line.
+ *
+ * Notes:
+ * <ul>
+ * <li>For optional fields null and unspecified are treated the same. Where
+ * appropriate this means null is mapped to a default value (e.g. empty map) or
+ * wrapped in an <code>Optional</code>.</li>
+ * </ul>
+ *
+ * @author Ryan Ascheman (rascheman at groupon dot com)
+ */
+public final class Version2g {
+
+    public String getId() {
+        return _id;
+    }
+
+    public DateTime getDate() {
+        return _date;
+    }
+
+    public Map<String, Element> getTimers() {
+        return _timers;
+    }
+
+    public Map<String, Element> getGauges() {
+        return _gauges;
+    }
+
+    public Map<String, Element> getCounters() {
+        return _counters;
+    }
+
+    public Map<String, String> getAnnotations() {
+        return _annotations;
+    }
+
+    public Map<String, String> getDimensions() {
+        return _dimensions;
+    }
+
+    public String getVersion() {
+        return _version;
+    }
+
+    private Version2g(final Builder builder) {
+        _id = builder._id;
+        _date = builder._date;
+        _dimensions = ImmutableMap.copyOf(builder._dimensions);
+        _annotations = ImmutableMap.copyOf(builder._annotations);
+        _version = builder._version;
+        _timers = builder._timers == null ? ImmutableMap.of() : ImmutableMap.copyOf(builder._timers);
+        _gauges = builder._gauges == null ? ImmutableMap.of() : ImmutableMap.copyOf(builder._gauges);
+        _counters = builder._counters == null ? ImmutableMap.of() : ImmutableMap.copyOf(builder._counters);
+    }
+
+    private final String _id;
+    private final DateTime _date;
+    private final ImmutableMap<String, String> _dimensions;
+    private final ImmutableMap<String, String> _annotations;
+    private final ImmutableMap<String, Element> _counters;
+    private final ImmutableMap<String, Element> _timers;
+    private final ImmutableMap<String, Element> _gauges;
+    private final String _version;
+
+    /**
+     * Builder for the Data class.
+     */
+    public static final class Builder extends OvalBuilder<Version2g> {
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            super(Version2g::new);
+        }
+
+        /**
+         * Sets the date field.
+         *
+         * @param value Value
+         * @return This builder
+         */
+        public Builder setDate(final DateTime value) {
+            _date = value;
+            return this;
+        }
+
+        /**
+         * Sets the id field.
+         *
+         * @param value Value
+         * @return This builder
+         */
+        public Builder setId(final String value) {
+            _id = value;
+            return this;
+        }
+
+        /**
+         * Sets the annotations field.
+         *
+         * @param value Value
+         * @return This builder
+         */
+        public Builder setAnnotations(final Map<String, String> value) {
```

Comment:
You should be able to accept an ImmutableMap directly thus avoiding the copy in the constructor.